DSCResources/TimezoneHelper.psm1

```diff
@@ -30,113 +38,146 @@ function New-TerminatingError
+
+        $TimeZone = (Get-CimInstance `
+            -ClassName WIN32_Timezone `
+            -Namespace root\cimv2).StandardName
+    }
 
     Write-Verbose -Message ($LocalizedData.CurrentTimezoneMessage `
         -f $Timezone)
 
-    return $TimeZone
-} # function Get-TimeZone
-
-function Get-TimeZoneId
-{
-    [CmdletBinding()]
-    param(
-        [Parameter(Mandatory = $true)]
-        [System.String]
-        $TimeZone
-    )
-    # Convert TimeZone Standard Name to TimeZone Id
     $timeZoneInfo = [System.TimeZoneInfo]::GetSystemTimeZones() |
         Where-Object StandardName -eq $TimeZone
 
     return $timeZoneInfo.Id
-}
+} # function Get-TimeZoneId
 
-function Test-TimeZone
+<#
+    .SYNOPSIS
+    Compare a timezone Id with the current timezone Id
+#>
+function Test-TimeZoneId
 {
     [CmdletBinding()]
     param(
         [Parameter(Mandatory = $true)]
         [System.String]
-        $ExpectTimeZoneId
+        $TimeZoneId
     )
     # Test Expected is same as Current
-    $currentId = Get-TimeZoneId -TimeZone (Get-TimeZone)
+    $currentTimeZoneId = Get-TimeZoneId
 
-    return $ExpectTimeZoneId -eq $currentId
-}
+    return $TimeZoneId -eq $currentTimeZoneId
+} # function Test-TimeZoneId
 
-function Set-TimeZone
+<#
+    .SYNOPSIS
+    Sets the current timezone using a timezone Id
+#>
+function Set-TimeZoneId
 {
     [CmdletBinding()]
     param(
         [Parameter(Mandatory = $true)]
         [System.String]
-        $TimeZone
+        $TimeZoneId
     )
-    $AddTypeCommand = Get-Command `
-        -Module Microsoft.Powershell.Utility `
-        -Name Add-Type `
+
+    $SetTimezoneCommand = Get-Command `
+        -Module Microsoft.PowerShell.Management `
+        -Name Set-Timezone `
         -ErrorAction SilentlyContinue
-    if ($AddTypeCommand)
-    {
-        # We can use Reflection to modify the TimeZone
-        Write-Verbose -Message ($LocalizedData.SettingTimezonedotNetMessage `
-            -f $TimeZone)
 
-        # Add the [TimeZoneHelper.TimeZone] type if it is not defined.
-        if (-not ([System.Management.Automation.PSTypeName]'TimeZoneHelper.TimeZone').Type)
-        {
-            Write-Verbose -Message ($LocalizedData.AddingSetTimeZonedotNetTypeMessage)
-            $SetTimeZoneCs = Get-Content `
-                -Path (Join-Path -Path $PSScriptRoot -ChildPath 'SetTimeZone.cs') `
-                -Raw
-            Add-Type `
-                -Language CSharp `
-                -TypeDefinition $SetTimeZoneCs
-        }
-        Set-TimeZoneUsingNET -Timezone $TimeZone
+    if ($SetTimezoneCommand)
+    {
+        Set-Timezone -Id $TimezoneId
     }
     else
     {
-        # For Nano Server the tzutil must be used
-        Write-Verbose -Message ($LocalizedData.SettingTimezoneTzUtilMessage `
-            -f $TimeZone)
-
-        try
+        $AddTypeCommand = Get-Command `
+            -Module Microsoft.Powershell.Utility `
+            -Name Add-Type `
```

Comment:
This block exists to load a particular module (and the cmdlet). Similar block exists for Timezone cmdlets also. Can we refactor this into a utility method so all the places can call this method instead?